src/entry.c

```diff
@@ -333,27 +320,38 @@ enum OperationReturnCode search_on_read(int rc, LDAPMessage *message, struct lda
 
                     if (!ld_entry)
                     {
-                        ld_error("search_on_read - out of memory - unable to create new entry!\n");
-
-                        return RETURN_CODE_FAILURE;
+                        ld_error("search_on_read - out of memory - unable to create new ld_entry_t!\n");
+                        if (entries)
+                        {
+                            while (entry_index)
+                            {
+                                --entry_index;
+                                ldap_memfree(entries[entry_index]);
```

Comment:
`ld_entry_t` defined in `entry_p.h` and has signature:
```
/*!
 * @brief ld_entry_t - Structure holds LDAP entry information.
 */
typedef struct ld_entry_s
{
    char* dn;                            //!< Distinguished name of the LDAP entry.
    GHashTable *attributes;              //!< Hash table with entry's attributes.
} ld_entry_t;
```

It is internal type of `libdomain` and therefore we cannot apply `ldap_memfree` to it.